docs/references/metadata.md

```diff
@@ -0,0 +1,273 @@
+        filepath = meta["source"]
+        with f = open(Path(filepath)):
+            content = f.read()
+            #...additional operations...
+        
+        self.provider.write_to_anchor("Output", batch)
+
+## Specifications
+
+For V2 plugins, metadata is stored in PyArrow tables as a Python
+dictionary in each column schema. But Arrow doesn't offer easy methods
+to access this metadata.
+The `ayx_python_sdk.core.utils` module provides
+functions to ease this access.
+
+### set_metadata
+
+`set_metadata(table, col_meta={}, schema=None)`
+
+This function updates column metadata from the given PyArrow table with
+a dictionary of column metadata or directly with a PyArrow Schema. It
+returns the new table with the new metadata. The input table remains
+unchanged.
+
+#### Example
+
+This example modifies the metadata of 2 columns,
+`volts` and `ampere`.
+
+    def on_record_batch(self, batch: "pa.Table", anchor: Anchor) - > None:
+        batch = set_metadata(batch, {
+            "volts": {
+                "description": "tension"
+            },
+            "ampere": {
+                "source": "https://something.com"
+            }
+        })
+
+    self.provider.write_to_anchor("Output", batch)
+
+### get_metadata
+
+`get_metadata(table, col_name)`
+
+Get all columns\' metadata or 1 column's metadata if
+`col_name` is given, from the input PyArrow table.
+
+-   If `col_name` is given, only the dictionary of
+    metadata for this column is returned.
+
+-   If `col_name` is not given, it returns a
+    dictionary of column names and their corresponding metadata
+    dictionary.
+
+#### Examples
+
+    def on_record_batch(self, batch: "pa.Table", anchor: Anchor) -> None:
+        meta_for_column_volts = get_metadata(batch, "volts")
+
+The above example `get_metadata` returns the
+metadata for 1 column, of the following format (in this example):
+
+    meta_for_column_volts = {
+        "type": 3,
+        "size": 2,
+        "scale": 0,
+        "source": "",
+        "description": ""
+    }
+
+###### Note
+You can see in the above example that a number represents the type.
+Please refer to the Types section below for the mapping between type ID
+and type names.
+
+
+This example returns all metadata:
+
+    def on_record_batch(self, batch: "pa.Table", anchor: Anchor) -> None:
+        all_meta = get_metadata(batch)
+
+The result dictionary looks like this:
+
+    all_meta = {
+        "volts": {
+            "type": 3,
+            "size": 2,
+            "scale": 0,
+            "source": "",
+            "description": ""
+        }
+        "device": {
+            "type": 9,
+            "size": 64,
+            "scale": 0,
+            "source": "",
+            "description": ""
+        }
+    }
+
+![Workflow Metadata Lifecycle Diagram](assets/workflow-metadata-lifecycle.png)
+
+You can find a more complete example at [Metadata Plugin
+Example](https://help.alteryx.com/developer-help/metadata-plugin-example).
```

Comment:
Point this to the new reference on ayx-developer-sdk repo.

This should point to the corresponding markdown version